src/app/lists/ListsPage.tsx

```diff
@@ -5,60 +5,92 @@ import { Schema } from "@/databaseUtils";
+        return newRow;
+    });
+    return formattedListData;
+};
+
 const ListsPage: React.FC<{}> = () => {
     const [isLoading, setIsLoading] = useState(true);
-    const [listsData, setListsData] = useState<ListRow[]>([]);
+    const [listData, setListData] = useState<ListRow[]>([]);
     const [comment, setComment] = useState("");
+    const [errorMessage, setErrorMessage] = useState<string | null>(null);
 
     useEffect(() => {
         (async () => {
             setIsLoading(true);
-            const fetchedData = await fetchData();
-            setListsData(
-                fetchedData.data.map((row) => {
-                    const newRow = {
-                        primaryKey: row.primary_key,
-                        rowOrder: row.row_order,
-                        itemName: row.item_name,
-                        ...Object.fromEntries(
-                            listsHeaderKeysAndLabels
-                                .filter(([key]) => /^\d+$/.test(key))
-                                .map(([key]) => [
-                                    key,
-                                    {
-                                        quantity:
-                                            row[`quantity_for_${key}` as keyof Schema["lists"]],
-                                        notes: row[`notes_for_${key}` as keyof Schema["lists"]],
-                                    },
-                                ])
-                        ),
-                    } as ListRow; // this cast is needed here as the type system can't infer what Object.fromEntries will return
-                    return newRow;
-                })
-            );
-            setComment(fetchedData.comment);
+            const listData = await fetchListData();
+            setListData(formatListData(listData));
+            setComment(listData.comment);
             setIsLoading(false);
         })();
     }, []);
 
+    useEffect(() => {
+        // This requires that the DB table has Realtime turned on
+        const subscriptionChannel = supabase
+            .channel("lists-table-changes")
+            .on("postgres_changes", { event: "*", schema: "public", table: "lists" }, async () => {
+                setErrorMessage(null);
+                try {
+                    const listData = await fetchListData();
+                    setListData(formatListData(listData));
+                } catch (error) {
+                    const logId = logErrorReturnLogId("Error with fetch: list data subscription", {
+                        error: error,
+                    });
+                    setListData([]);
+                    setErrorMessage(`Error fetching data, please reload. Log ID: ${logId}`);
+                }
+            })
+            .subscribe((status, err) => {
+                subscriptionStatusRequiresErrorMessage(status, err, "website_data") &&
+                    setErrorMessage("Error fetching data, please reload");
```

Comment:
I think this will be more readable than using the logical AND operator
```suggestion
                if (subscriptionStatusRequiresErrorMessage(status, err, "website_data")) {
                    setErrorMessage("Error fetching data, please reload");
                }
```